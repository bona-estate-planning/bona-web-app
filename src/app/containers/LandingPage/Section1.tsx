/**
 *
 * Section1
 *
 */
import { useMediaQuery, useTheme } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { memo } from 'react';
import { Link } from 'react-router-dom';
import Image1 from './image1.svg';
import Image2 from './image2.svg';

interface Props {}

const useStyles = makeStyles(theme => {
  return {
    section: (isMd: boolean) => ({
      paddingTop: theme.spacing(isMd ? 10 : 3.5),
      textAlign: 'center',
    }),
    headline: {
      fontWeight: theme.typography.fontWeightBold,
      fontSize: theme.spacing(5),
      marginBottom: theme.spacing(3),
    },
    subtitle: {
      marginBottom: theme.spacing(3),
    },
    button: {
      borderRadius: 99999,
      padding: theme.spacing(1, 10),
      textTransform: 'none',
      fontWeight: theme.typography.fontWeightBold,
    },
    imageBox: (isMd: boolean) => ({
      marginTop: theme.spacing(isMd ? -4 : -2),
      height: isMd ? '30vw' : '35vw',
      minHeight: 20,
      backgroundImage: `url(${Image1}), url(${Image2})`,
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'bottom left, bottom right',
      backgroundSize: 'contain',
    }),
  };
});

export const Section1 = memo((props: Props) => {
  const theme = useTheme();
  const isMd = useMediaQuery(theme.breakpoints.up('md'));
  const classes = useStyles(isMd);

  return (
    <Box pt={isMd ? 10 : 3.5} className={classes.section}>
      <Container>
        <Typography component="h2" className={classes.headline}>
          Estately
        </Typography>
        <Typography component="p" variant="h5" className={classes.subtitle}>
          Simple estate planning for everyone.
        </Typography>
        <Button
          component={Link}
          to="/wizard"
          variant="contained"
          color="primary"
          className={classes.button}
        >
          Start a plan
        </Button>
      </Container>
      <Box className={classes.imageBox} />
    </Box>
  );
});
