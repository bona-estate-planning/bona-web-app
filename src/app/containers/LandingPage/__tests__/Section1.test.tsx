import React from 'react';
import { render } from '@testing-library/react';
import { Section1 } from '../Section1';
import { BrowserRouter } from 'react-router-dom';

describe('<Section1  />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Section1 />
      </BrowserRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
