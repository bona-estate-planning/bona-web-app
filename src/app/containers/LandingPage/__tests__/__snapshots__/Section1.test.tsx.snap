// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section1  /> should match snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-8 makeStyles-section-1 makeStyles-section-6"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="MuiTypography-root makeStyles-headline-2 MuiTypography-body1"
    >
      Estately
    </h2>
    <p
      class="MuiTypography-root makeStyles-subtitle-3 MuiTypography-h5"
    >
      Simple estate planning for everyone.
    </p>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-4 MuiButton-containedPrimary"
      href="/wizard"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        Start a plan
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiBox-root MuiBox-root-9 makeStyles-imageBox-5 makeStyles-imageBox-7"
  />
</div>
`;
